### PR TITLE
[SPARK-26963][MLLIB] SizeEstimator can't make some JDK fields accessible in Java 9+

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Don't use inaccessible fields in SizeEstimator, which comes up in Java 9+

## How was this patch tested?

Manually ran tests with Java 11; it causes these tests that failed before to pass.
This ought to pass on Java 8 as there's effectively no change for Java 8.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/util/SizeEstimator.scala [hunk 1]
```diff
@@ -334,9 +334,21 @@ object SizeEstimator extends Logging {
...
+            field.setAccessible(true) // Enable future get()'s on this field
+            pointerFields = field :: pointerFields
+          } catch {
+            // If the field isn't accessible, we can still record the pointer size
+            // but can't know more about the field, so ignore it
+            case _: SecurityException =>
+              // do nothing
+            // Java 9+ can throw InaccessibleObjectException but the class is Java 9+-only
+            case re: RuntimeException
+                if re.getClass.getSimpleName == "InaccessibleObjectException" =>
```
Comment (imatiach-msft) [inline core/src/main/scala/org/apache/spark/util/SizeEstimator.scala]: minor comment/question: if it's not an InaccessibleObjectException, wouldn't it throw a case match error which might be confusing?  Just wondering if a default catch-all should re-raise the original "re: RuntimeException"

**Review by imatiach-msft — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/util/SizeEstimator.scala [hunk 2]
```diff
@@ -334,9 +334,21 @@ object SizeEstimator extends Logging {
         if (fieldClass.isPrimitive) {
           sizeCount(primitiveSize(fieldClass)) += 1
         } else {
-          field.setAccessible(true) // Enable future get()'s on this field
+          // Note: in Java 9+ this would be better with trySetAccessible and canAccess
+          try {
+            field.setAccessible(true) // Enable future get()'s on this field
+            pointerFields = field :: pointerFields
```
[the remark this answers is not in the document]
Comment (srowen) [reply core/src/main/scala/org/apache/spark/util/SizeEstimator.scala]: No, the point is that we don't want to try to access it below; it's not accessible

File: core/src/main/scala/org/apache/spark/util/SizeEstimator.scala (hunk 1 above)
Comment (srowen) [reply core/src/main/scala/org/apache/spark/util/SizeEstimator.scala]: The exception would just propagate, as before, if it's not matched by either case. That's what we want, if something else goes wrong.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**